### PR TITLE
OOP school library: set up associations.

### PR DESCRIPTION
**Pull request by habasm**
## Description
In this project, we are going to finish creating the remaining classes for our school library and create the associations between them.

### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @habasm, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linter passed ✔️ 
- Readme file professional ✔️
- Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._
- [ ] While you have done a good job, you must have forgotten to work on the Person class, I would advise you to add the rentals to the instance variable attr_accessor in the Person class so that it can be read and writable. You will need to initialize it also
```
...
def initialize(age, name = "Unknown", parent_permission = true)
...
@rentals = []
...
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

## Optional suggestions
- I would have invalidated this work but the `NOT AVAILABLE` that you add to the live demo section made me go on, However, I will advise you to remove the whole link from the readme file so that it would not lead visitors to a malicious site.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative 'rental'
+class Book
+  attr_accessor :title, :author
+
+  attr_reader :rental
```
[the remark this answers is not in the document]
Comment (habasm) [reply book.rb]: @Olamarx fixed

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require_relative 'rental'
+class Book
+  attr_accessor :title, :author
+
+  attr_accessor :rental
```
Comment (rloterh) [inline book.rb:5]: - [ ] [OPTIONAL]
Let's optimize our code by defining all attr_accessor on one line, like below:
```ruby
class Book
  attr_accessor :title, :author, :rental
  ...
end
```

**Comment by habasm** [comment]
@rloterh, so what's the decision? I didn't get you. Thanks.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Habitamu,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Good job establishing object associations!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- Kindly remember to always remove demo link from ReadMe if the app is not yet deployed!
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rloterh** [comment]
> @rloterh, so what's the decision? I didn't get you. Thanks.

@habasm 
Kindly refer the review comments above. 

Cheers! 😃

**Pull request merged**